tests/providers/openlineage/plugins/test_listener.py

```diff
@@ -67,12 +68,16 @@ def render_df():
 def test_listener_does_not_change_task_instance(render_mock, xcom_push_mock):
     render_mock.return_value = render_df()
 
-    dag = DAG(
-        "test",
-        start_date=dt.datetime(2022, 1, 1),
-        user_defined_macros={"render_df": render_df},
-        params={"df": render_df()},
-    )
+    with pytest.warns(
+        RemovedInAirflow3Warning,
+        match="The use of non-json-serializable params is deprecated and will be removed in a future release",
```

Comment:
Why airflow 3?
Providers code should not raise Airflow3 warnings
Isn't this test already covered by ?
https://github.com/apache/airflow/blob/57b12fe0ce14487188da2c12a7bbfa1108008301/tests/serialization/test_dag_serialization.py#L933-L945

Reply:
It's covered in core tests. So, I will update the test case.